contribs/gmf/src/directives/drawfeature.js

```diff
@@ -0,0 +1,399 @@
+      ol.Object.getChangeEventType(ol.interaction.InteractionProperty.ACTIVE),
+      this.handleActiveChange_, this);
+
+
+  // === AZIMUT / CIRCLE ===
+
+  helpMsg = gettext('Click to start drawing azimut');
+  contMsg = gettext('Click to finish');
+
+  /**
+   * @type {ngeo.interaction.MeasureAzimut}
+   * @export
+   */
+  this.measureAzimut = new ngeo.interaction.MeasureAzimut({
+    style: new ol.style.Style(),
+    startMsg: $compile('<div translate>' + helpMsg + '</div>')($scope)[0],
+    continueMsg: $compile('<div translate>' + contMsg + '</div>')($scope)[0]
+  });
+
+  var measureAzimut = this.measureAzimut;
+  this.interactions_.push(measureAzimut);
+  measureAzimut.setActive(false);
+  ngeoDecorateInteraction(measureAzimut);
+  this.map.addInteraction(measureAzimut);
+  ol.events.listen(measureAzimut, ngeo.MeasureEventType.MEASUREEND,
+      /**
+       * @param {ngeo.MeasureEvent} event Event.
+       */
+      function(event) {
+        // In the case of azimut measure interaction, the feature's geometry is
+        // actually a collection (line + circle)
+        // For our purpose here, we only need the circle.
+        var geometry = /** @type {ol.geom.GeometryCollection} */
+            (event.feature.getGeometry());
+        event.feature = new ol.Feature(geometry.getGeometries()[1]);
+        this.handleDrawEnd_(ngeo.GeometryType.CIRCLE, event);
+      }, this);
+  ol.events.listen(measureAzimut,
+      ol.Object.getChangeEventType(ol.interaction.InteractionProperty.ACTIVE),
+      this.handleActiveChange_, this);
+
+
+  // === RECTANGLE ===
+
+  /**
+   * @type {ol.interaction.Draw}
+   * @export
+   */
+  this.drawRectangle = new ol.interaction.Draw({
+    type: ol.geom.GeometryType.LINE_STRING,
+    geometryFunction: function(coordinates, geometry) {
+      if (!geometry) {
+        geometry = new ol.geom.Polygon(null);
+      }
+      var start = coordinates[0];
+      var end = coordinates[1];
+      geometry.setCoordinates([
+        [start, [start[0], end[1]], end, [end[0], start[1]], start]
+      ]);
+      return geometry;
+    },
+    maxPoints: 2
+  });
+
+  var drawRectangle = this.drawRectangle;
+  this.interactions_.push(drawRectangle);
+  drawRectangle.setActive(false);
+  ngeoDecorateInteraction(drawRectangle);
+  this.map.addInteraction(drawRectangle);
+  ol.events.listen(drawRectangle, ol.interaction.DrawEventType.DRAWEND,
+      this.handleDrawEnd_.bind(this, geomType.RECTANGLE), this);
+  ol.events.listen(drawRectangle,
+      ol.Object.getChangeEventType(ol.interaction.InteractionProperty.ACTIVE),
+      this.handleActiveChange_, this);
+
+
+  // === TEXT ===
+
+  /**
+   * @type {ol.interaction.Draw}
+   * @export
+   */
+  this.drawText = new ol.interaction.Draw({
+    type: ol.geom.GeometryType.POINT
+  });
+
+  var drawText = this.drawText;
+  this.interactions_.push(drawText);
+  drawText.setActive(false);
+  ngeoDecorateInteraction(drawText);
+  this.map.addInteraction(drawText);
```

Comment:
```js
  interaction.setActive(false);
  ngeoDecorateInteraction(interaction);
  this.map.addInteraction(interaction);
```

maybe this could be factorized in a `this.interactions_.forEach()` ?
It's a bit heavy.